README.md

```diff
@@ -1,63 +1,91 @@
 
-By leveraging this library, smart contract developers can leverage on verifiable randomness on-chain that is tamper-proof and bias resistant.
 
-This library is designed with modularity and simplicity in mind, allowing developers to extend and integrate it into their existing projects easily.
+## Overview
+In decentralized applications, unpredictable random values are essential for ensuring fairness and preventing manipulation. Secure randomness ensures that outcomes are both **unbiased** and **unpredictable**, making it impossible for any single participant to influence the result. This is particularly critical in use cases such as NFT trait generation, decentralized governance, lotteries, and other scenarios where fairness is a core requirement.
 
+This repository provides a Solidity-based library for requesting on-chain randomness powered by the **Dcipher threshold network**. Random values are generated through supported threshold signature schemes, which guarantees that the output is publicly verifiable, tamper-resistant, and unpredictable until revealed. This cryptographic foundation ensures that smart contracts relying on randomness can operate in a secure and trust-minimized manner.
 
-### Smart Contract Addresses
+The library is designed with modularity and simplicity in mind, allowing developers to easily integrate it into their existing smart contract projects. Its extensible architecture makes it suitable for a wide range of applications that require robust on-chain randomness.
 
-| Contract        | Address | Network          |
-|-----------------|---------|------------------|
-| SignatureSender Proxy |  0x1c86A81D3CDD897aFdcA62a9b7219a39Aef7910B  | Filecoin Calibration Testnet |
-| SignatureSender Implementation | 0x1790de5a9fBA748DCAf05e3a1755Cf1DD6b9B0F8   | Filecoin Calibration Testnet |
-| RandomnessSender Proxy |  0x9c789bc7F2B5c6619Be1572A39F2C3d6f33001dC  | Filecoin Calibration Testnet |
-| RandomnessSender Implementation |  0xF684f13850932bC7B51bd6bFF9236FB19E55F2B1  | Filecoin Calibration Testnet |
-| SignatureSchemeAddressProvider |  0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A  | Filecoin Calibration Testnet |
-| MockBN254SignatureScheme | 0xE5aedc08Cf2B5650Cd84CE6DcaDC3763bAa8770B   | Filecoin Calibration Testnet |
-| MockRandomnessReceiver |  0x6e7B9Ccb146f6547172E5cef237BBc222EC4D676  | Filecoin Calibration Testnet |
+### Features
+Powered by the Dcipher threshold network and various signature schemes, this randomness libaries offers:
 
+- **Verifiable & Trustless**: Not just decentralized, unpredictrable, but also publickly verifiable trustless randomness.
+- **Unpredictable**: Using highly unpredictable and diverse inputs for a hash function like `keccak256()`.
```

Comment:
I don't really understand what the relationship to keccak256 is. We should either be info-light (predictability in line with chance) or precise (uses keccak256's avalanche effect to remove algebraic structure)

Reply:
I think we can use info-light in this case since it is an intro. Not necessarily need to be detailed.